Remove deprecated rich_compare metadata

mdickinson opened the pull request:
This PR removes the deprecated `rich_compare` metadata.

Closes #602.

mdickinson commented:
There are some unused imports that need fixing.

corranwebster reviewed (approved):
Looks good.

I'd suggest perhaps renaming `test_rich_compare.py` and the classes inside it which refer to "rich compare" to use something like "equality compare", "comparison mode" or something similar as appropriate, since there is now no other reference to "rich comparison" in the codebase.

mdickinson commented:
> I'd suggest perhaps renaming `test_rich_compare.py` and the classes inside it

Done; I've cleaned up that test module; while I was at it, I added missing tests for `ComparisonMode.none` complementing the existing ones for `ComparisonMode.identity` and `ComparisonMode.equality`.
